lambda-buffers-proto-compat/src/LambdaBuffers/ProtoCompat/ProtoCompat.hs

```diff
@@ -0,0 +1,562 @@
+{-# LANGUAGE KindSignatures, DuplicateRecordFields #-}
+{-# LANGUAGE DeriveGeneric, TypeSynonymInstances #-}
+{-# LANGUAGE DataKinds, OverloadedLabels #-}
+{-# OPTIONS_GHC -Wno-missing-kind-signatures #-}
+{-# OPTIONS_GHC -Wno-missing-deriving-strategies #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+module LambdaBuffers.ProtoCompat.ProtoCompat where
+
+import Data.List.NonEmpty (NonEmpty, nonEmpty)
+import Data.Text (Text)
+import qualified Proto.Compiler as P
+import qualified Proto.Compiler_Fields as P
+import Data.ProtoLens (defMessage)
+import Data.Kind (Type)
+import GHC.Natural (Natural)
+import Control.Lens ( (&), (^.), to, (.~), Getter )
+import Data.Generics.Labels ()
+import GHC.Generics (Generic)
+import Data.Foldable (toList)
+import qualified Data.Text as T
+
+note :: e -> Maybe a -> Either e a
+note e = \case
+  Nothing -> Left e
+  Just a  -> Right a
+
+-- something like this probably exists in lens but i can't find it
+traversing :: (Applicative f,  Traversable t) => (a -> f b) -> Getter (t a) (f (t b))
+traversing f = to $ \ta -> traverse f ta
+
+
+data SourceInfo = SourceInfo {
+  file    :: Text,
+  posFrom :: SourcePosition,
+  posTo   :: SourcePosition
+} deriving (Show, Eq, Ord, Generic)
+
+data SourcePosition = SourcePosition {
+  column :: Int,
+  row    :: Int
+} deriving (Show , Eq, Ord, Generic)
+
+-- for our purposes we really only need a single name type
+data LBName
+  = LBName {name :: Text, sourceInfo :: SourceInfo}
+     deriving (Show, Eq, Ord, Generic)
```

Comment:
Wdym?

Reply:
The different names are the result of different parsing/validation functions. The fact that they share the underlying Text is not making them the 'same' thing.